### PR TITLE
Dev/verification mail

### PR DESCRIPTION
**Pull request by shivamgoyal7**
### Functionality:
user varification through mail

### Solution:
Mail is sent to all the members to verify user authenticate.

### Risk level:
- [x] high 
- [ ] medium
- [ ] low

### How to test:
firebase deploy

### CODE REVIEW

**Comment by viv9k** [comment]
I want demo on this today evening

**Comment by viv9k** [comment]
also change the target branch to Adding-organization-feature

**Review by viv9k — CHANGES_REQUESTED** [review]
Please complete the required changes

File: functions/createNewOrgAPI.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const functions = require("firebase-functions");
 const cors = require("cors")({ origin: true });
 
 const admin = require("firebase-admin");
+const sendEmail = require("./sendEmailAPI.js");
```
[the remark this answers is not in the document]
Comment (shivamgoyal7) [reply functions/createNewOrgAPI.js:10]: done

File: functions/createNewOrgAPI.js [hunk 2]
```diff
@@ -123,6 +124,11 @@ exports.createNewTeamWithLabels = functions.https.onRequest((request, response)
                 return Promise.resolve(teamData);
             }
         });
+
+        teamMembers.forEach((element) => {
+            sendEmail.sendVerificationEmail(teamName, teamManagerEmail, teamDescription, element, organizationDomain);
```
[the remark this answers is not in the document]
Comment (shivamgoyal7) [reply functions/createNewOrgAPI.js:129]: done

File: functions/sendEmailAPI.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    },
+  });
+  return Promise.resolve(sendEmailPromise).then(() => {
+    console.log("Verification Email Sent Successfully!");
+    return 0;
+  })
+      .catch((error) => {
+        console.error("Error in Sending Verification Email: ", error);
+      });
+};
```
[the remark this answers is not in the document]
Comment (shivamgoyal7) [reply functions/sendEmailAPI.js:24]: done

File: src/app/app-routing.module.ts [hunk 4]
```diff
@@ -22,7 +23,8 @@ const routes: Routes = [
   { path: '', component: MyDashBoardComponent },
   { path: 'CreateNewOrganization', component: CreateNewOrganizationComponent },
   { path: 'patch', component: PatchComponent },
-  { path: "login", component: LoginComponent, canActivate: [AngularFireAuthGuard], data: { authGuardPipe: redirectLoggedInToDashboard } }
+  { path: "login", component: LoginComponent, canActivate: [AngularFireAuthGuard], data: { authGuardPipe: redirectLoggedInToDashboard } },
+  { path: 'verifyUser/:organizationDomain/:teamName/:userEmail', component: UserVerificationComponent, canActivate: [AngularFireAuthGuard] },
```
[the remark this answers is not in the document]
Comment (shivamgoyal7) [reply src/app/app-routing.module.ts:27]: yes it is working but it redirects to dashboard so I have removed it and use condition to handle

[the remark this answers is not in the document]
Comment (viv9k) [reply src/app/app-routing.module.ts:27]: You can use something like: const redirectLoggedInToDashboard = () => redirectLoggedInTo(['']);

const redirectLoggedInToVerifyUser = () => redirectLoggedInTo(['verifyUser/orgDomain/teamName/userEmail']);

then add => data: { authGuardPipe: redirectLoggedInToVerifyUser }

**Review by viv9k — CHANGES_REQUESTED** [review]
Please complete the required changes

**Comment by shivamgoyal7** [comment]
changes are not coming in this pr

**Comment by shivamgoyal7** [comment]
so im closing this pr

**Pull request closed without merging**